code/game/objects/items/pinpointer.dm

```diff
@@ -152,6 +152,74 @@
 	if(!target) //target can be set to null from above code, or elsewhere
 		active = FALSE
 
+/obj/item/pinpointer/tracker //A modified pinpointer that tracks mobs with tracking implants
+	name = "tracking implant pinpointer"
+	desc = "A handheld tracking device that points to individuals with active tracking implants."
+	icon_state = "pinpointer_tracker"
+	var/has_owner = FALSE
+	var/pinpointer_owner = null
+
+/obj/item/pinpointer/tracker/proc/implanted(mob/living/H)
+	var/turf/here = get_turf(src)
+	for(var/obj/item/implant/I in H.implants)
+		if(istype(I, /obj/item/implant/tracking))
+			continue
+		else
+			return FALSE
+
+	if(H.z == 0 || H.z == here.z)
+		var/turf/there = get_turf(H)
+		return (H.z != 0 || (there && there.z == here.z))
+
+	return FALSE
+
+/obj/item/pinpointer/tracker/attack_self(mob/living/user)
+	if(active)
+		toggle_on()
+		user.visible_message(span_notice("[user] deactivates [user.p_their()] pinpointer."), span_notice("You deactivate your pinpointer."))
+		return
+
+	if (has_owner && !pinpointer_owner)
+		pinpointer_owner = user
+
+	if (pinpointer_owner && pinpointer_owner != user)
+		to_chat(user, span_notice("The pinpointer doesn't respond. It seems to only recognise its owner."))
```

Comment:
isn't a , better here?

Reply:
On the string?

Reply:
yes

Reply:
technically a semicolon which I'll do since the comma would lack a conjunction